### ISSUE TITLE
Ability to disable taint propagators for operators.

### ISSUE THREAD

**Issue by jcttk** [semgrep/semgrep#9767]
**Describe the solution you'd like**
Is there going to be support for disabling taint propagation through operators? like there is for 

taint_assume_safe_functions
taint_assume_safe_indexes

**Comment by ievans** [comment]
hey @jcttk this question may be better asked on our Slack for a fast response https://go.semgrep.dev/slack

@nmote any thoughts?

**Comment by nmote** [comment]
What's the use case here? We have options that are something like "taint_assume_safe_boolean" and "taint_assume_safe_numbers" (off the top of my head, might be slightly different) which use type inference to avoid propagating taint through expressions such as operators, when appropriate. For example, see this test case:

* https://github.com/semgrep/semgrep/blob/develop/tests/rules/taint_assume_safe_booleans1.java
* https://github.com/semgrep/semgrep/blob/develop/tests/rules/taint_assume_safe_booleans1.yaml

Would this work for you?

**Comment by jcttk** [comment]
Yep so a use case:

I don't want to propagate taint if my source is string concatenated.

**Comment by nmote** [comment]
Makes sense. Unfortunately I don't think we have anything explicit like that at the moment. You could see about adding a sanitizer that matches expressions of the form `$X + $Y`.  Either way, let's leave this open since it sounds like the kind of thing we might implement in the future.

**Comment by chejn** [comment]
> Makes sense. Unfortunately I don't think we have anything explicit like that at the moment. You could see about adding a sanitizer that matches expressions of the form `$X + $Y`. Either way, let's leave this open since it sounds like the kind of thing we might implement in the future.

this worked thanks :)
